### ISSUE TITLE
[RFC] Package list enhancements

### ISSUE THREAD

**Issue by nbehier** [bolt/marketplace-site#59]
Every list page could be enhanced. We could use a [JS script](https://www.dynatable.com/) to easily filter/search extensions/themes list. Mockup attached to launch discussion.
![boltstorelistpage](https://cloud.githubusercontent.com/assets/1059917/16797108/433becaa-48e6-11e6-8afe-504657af440a.PNG)

**Comment by GawainLynch** [comment]
This should be a lot easier to do with the work completed recently, and also @rossriley is going to have a look at the search functions soon too.

@nbehier should we be able to agree (I stay out of UI, but will get buy-in from the others), are you interested in tackling some of the UI work?

ping @PhillippOhlandt

**Comment by nbehier** [comment]
If I can help, I'll be happy to contribute!

**Comment by PhillippOhlandt** [comment]
Hmm,

I wouldn't start replacing single components in the store yet. Otherwise, every page will look differently. We should hold stuff like that back to a major redesign and restructuring of the store.

**Comment by GawainLynch** [comment]
Connecting to #56 then

**Comment by GawainLynch** [comment]
@PhillippOhlandt Yes, but we can try and redo everything, and never get it done, or we could be a little strategic and get bits started now.

I agree that it should stay consistent though.

**Comment by nbehier** [comment]
@PhillippOhlandt I'm totally agree with you, list page and homepage redesign have to serve the same goal => store redesign. Do you want to sum up all ideas on #56 or a new global issue ?

**Comment by PhillippOhlandt** [comment]
@GawainLynch the new store homepage structure would still have the same (ugly :D) style. The proposed listview above can't be done with semantic UI in that form. There are also some things like "downloads" and "starred" that will fuck performance at the current state.

**Comment by PhillippOhlandt** [comment]
@nbehier when you show me a "non-mockup" without downloads and starred counts, we can see if it can be already added to the store yet.

**Comment by PhillippOhlandt** [comment]
@nbehier please dont get me wrong. i like the idea of improved listviews because the current ones arent optimal at all. when we want to improve them right now, we need to find something in the middle of "current" and "perfect".

**Comment by GawainLynch** [comment]
> There are also some things like "downloads" and "starred" that will #### performance at the current state.

Yeah, but lets iterate rather than try to land perfect now … too much to do and not enough time, and if people like @nbehier are willing to help with bits then I say let's at lest get it moving in a branch

**Comment by PhillippOhlandt** [comment]
@GawainLynch i agree, read my last two replies :)

**Comment by GawainLynch** [comment]
> @GawainLynch i agree, read my last two replies :)

I was typing while you were too :laughing:

**Comment by nbehier** [comment]
Another way to display search/sort/filters could be with a sidebar.
![boltstorelistpage2](https://cloud.githubusercontent.com/assets/1059917/16803364/4f1d2320-4906-11e6-86fb-522195177935.PNG)

**Comment by PhillippOhlandt** [comment]
Could be doable but please with a list view and not a grid view for extensions. Could you please look if such a layout is possible with SemanticUI (which we use at the current store)?

**Comment by nbehier** [comment]
I think we could use : 
- For extension : first example of [cards](http://semantic-ui.com/views/card.html#cards) with small icon (and [with link version](http://semantic-ui.com/views/card.html#link-card) );
- For Bolt version badge : [label](http://semantic-ui.com/elements/label.html#label) template ;
- For Sidebar : [Sidebar module](http://semantic-ui.com/modules/sidebar.html), [Vertical menu](http://semantic-ui.com/collections/menu.html#vertical-menu) or simple form in grid column.

**Comment by PhillippOhlandt** [comment]
please no cards for the listview.

**Comment by nbehier** [comment]
OK, so we can use [list with image](http://semantic-ui.com/elements/list.html#image) as current page...
I don't really understand difference with cards... `<div/>` for both anyway. What do you have in mind, just a preference to list-view or something else ?

**Comment by PhillippOhlandt** [comment]
thats not the type of list we need. The current list view isn't that bad. it just needs the sidebar and some new values. 

Why not cards: because 99% of all extensions don't have an image and we would need the cards in a grid, which is suboptimal here.

**Comment by navelpluisje** [comment]
@PhillippOhlandt I think with the latest extensions more then 50% has an image. On the other hand this could be a nice way to push submitters a bit more to use images. Beside that I think the cards or item(2 columns) of semantic UI are a nicer way to show the extensions then a list, especially on a larger screen. 
Or why don't we let the user choose? Let them toggle list/card view. By using all the items from the same library everything will look pretty consistent, so no hassle I think.

**Comment by PhillippOhlandt** [comment]
I am not sure if we should change something now. The store needs a redesign and it is very likely that we don't stick with semantic ui.

**Comment by navelpluisje** [comment]
Fair enough, but is there a timeline, or anything more specific?

**Comment by PhillippOhlandt** [comment]
Not really. I am mostly waiting for the styleguide and the Bolt backend CSS refactor.
